### PR TITLE
Add Poisson documentation

### LINKED ISSUE #26732: Poisson documentation
Documentation should be added for the Poisson cost function PR#25686, issue #14490 should have the maths.

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
Adds Possion documentation

**To test:**
Build docs and read through and check it includes all the information it needs

Fixes #26732 

*This does not require release notes* because **its documentation**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/fitting/fitcostfunctions/Poisson.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+=====================
+Poisson Cost Function
+=====================
+
+.. index:: Poisson
+
+Description
+-----------
+
+The Poisson cost function is designed to be applied to data which has a Poisson distribution and a low number of counts.
```
Comment (AnthonyLim23) [inline docs/source/fitting/fitcostfunctions/Poisson.rst]:
```suggestion
The Poisson cost function is designed to be applied to data, which has a low number of counts.
```

**Review by AnthonyLim23 — DISMISSED** [review]

**Comment by martyngigg** [comment]
There was a mix up with `release-next` yesterday. I'm currently fixing up this branch.

**Comment by martyngigg** [comment]
I've rebase this onto the current HEAD of `release-next`. Could you just double check that it all looks as you expect?

**Review by martyngigg — COMMENTED** [review]
Was `docs/source/algorithms/ComputeIncoherentDOS-v1.rst` meant to be deleted?

**Comment by alicerussell1** [comment]
I didn't mean to remove the file so I've added it back in. The other changes look as expected after the rebase, should be ready now.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**